src/main/kotlin/org/rust/lang/refactoring/RsNamesValidator.kt

```diff
@@ -18,10 +19,10 @@ class RsNamesValidator : NamesValidator {
         isIdentifier(name, project, true)
 
     fun isIdentifier(name: String, @Suppress("UNUSED_PARAMETER") project: Project?, withPrimitives: Boolean): Boolean =
-        if (withPrimitives) {
-            getLexerType(name) == IDENTIFIER && name !in PrimitiveTypes
-        } else {
-            getLexerType(name) == IDENTIFIER
+        when (getLexerType(name)) {
+            IDENTIFIER -> !withPrimitives || name !in PrimitiveTypes
+            QUOTE_IDENTIFIER -> name !in PredefinedLifetimes
```

Comment:
Yeah, renaming to `'static` should be allowed. And its declaration could be removed after renaming to `'static` with help of `RenamePsiElementProcessor`, that's a good idea.